src/runtime/lpython_intrinsic_numpy.py

```diff
@@ -410,3 +410,23 @@ def ceil(x: f32) -> f32:
     if x <= f32(0) or x == resultf:
         return resultf
     return resultf + f32(1)
+
+########## trunc ##########
+
+@ccall
+def _lfortran_dtrunc(x: f64) -> f64:
+    pass
+
+@overload
+@vectorize
+def trunc(x: f64) -> f64:
+    return _lfortran_dtrunc(x)
+
+@ccall
+def _lfortran_strunc(x: f32) -> f32:
+    pass
+
+@overload
+@vectorize
+def trunc(x: f32) -> f32:
+    return _lfortran_strunc(x)
```

Comment:
This shouldn't be needed I think.

Reply:
Yup, let's remove this.

Reply:
@khushi-411 can you please remove this change? I think it's not needed, and I think it will also bring down the amount of changes in reference tests, since the symbol table IDs should stay unchanged.

Reply:
After removing this overload, the tests fail for the `fp32` case. The other case might be to obliterate it (but this is not even an option to be considered).

Therefore, I think we'll need this. Please correct me if I'm wrong; if lpython follows the numpy/array-api convention, `np.trunc` supports `fp32`, so I think we should add an overload for it. (And I guess for complex dtypes too...)

why do you think we should remove it?

Reply:
I would have to see the failure. Yes, we need to support all types and tests. The way it should work is that the frontend creates an IntrinsicFunction node, and then the LLVM backend calls an ASR->ASR pass where this node gets implemented, or it might even later implement it directly.

Reply:
Interesting, will search about it more.

About the failures, I got:
```diff
semantic error: Arguments do not match for any generic procedure, trunc
  --> /home/khushi/Documents/lpython/integration_tests/elemental_13.py:54:16
   |
54 |     observed = trunc(arraynd)
   |                ^^^^^^^^^^^^^^ 
```
In this part of the code:
```python
    arraynd: f32[32, 16, 8, 4] = empty((32, 16, 8, 4), dtype=float32)

    for i in range(32):
        for j in range(16):
            for k in range(8):
                for l in range(4):
                    arraynd[i, j, k, l] = f32(f64((-1)**l) * sqrt(float(i + j + j + l)))

    observed: f32[32, 16, 8, 4] = empty((32, 16, 8, 4), dtype=float32)
    observed = trunc(arraynd)
```

Reply:
I think the code in the ast->asr has to be improved to take care of this, like it does for sin/cos.

Reply:
I think `sin/cos`, too, uses `fp32` overloads as I did here; ref:
https://github.com/lcompilers/lpython/blob/main/src/runtime/lpython_intrinsic_numpy.py#L14-#L21

when I removed those overloads, I got the same errors as above. Please see:
```python
semantic error: Arguments do not match for any generic procedure, sin
  --> /home/khushi/Documents/lpython/integration_tests/elemental_01.py:10:24
   |
10 |         assert abs(sin(sin(array[i])) - result[i]) <= eps
   |                        ^^^^^^^^^^^^^ 
```

Reply:
I see, I didn't realize that. Ok then, we can leave it then, and fix it later.